nessie/src/main/java/org/apache/iceberg/nessie/NessieTableOperations.java

```diff
@@ -92,6 +92,18 @@ private TableMetadata loadTableMetadata(String metadataLocation, Reference refer
     // Nessie CLI will provide a reference aware GC functionality for the expired/unreferenced
     // files.
     newProperties.put(TableProperties.GC_ENABLED, "false");
+
+    boolean metadataCleanupEnabled =
+        newProperties
+            .getOrDefault(TableProperties.METADATA_DELETE_AFTER_COMMIT_ENABLED, "false")
+            .equalsIgnoreCase("true");
+    if (metadataCleanupEnabled) {
+      newProperties.put(TableProperties.METADATA_DELETE_AFTER_COMMIT_ENABLED, "false");
+      LOG.warn(
+          "Clean up of table metadata files during commit is disabled as it can be in use by other references."
+              + " Use nessie-gc tool for reference aware GC");
```

Comment:
> I guess the WARN is going to be logged on every load, unless the user permanently resets the METADATA_DELETE_AFTER_COMMIT_ENABLED property to false, right?

No, this will be set in table metadata refresh flow. So, this property will be set to false and written to file in the first refresh.  So, it won't print on every load. For example, this testcase execution prints it only once.

Reply:
> How about Automatic table metadata files cleanup was requested, but disabled because the Nessie catalog can use historical metadata files. Please set the " + TableProperties.METADATA_DELETE_AFTER_COMMIT_ENABLED + " property to 'false'. Use the 'nessie-gc' tool for history-aware GC".?

I don't think user has to reset it again (based on my above comment). So, I am thinking to update like this.

`Automatic table metadata files cleanup was requested, but disabled because the Nessie catalog can use historical metadata files from other references. Use the 'nessie-gc' tool for history-aware GC".`

Reply:
SGTM :+1: